addons/mail/models/mail_message.py

```diff
@@ -978,24 +1006,34 @@ def _to_store(self, store, /, *, fields=None, format_reply=True, msg_vals=None,
                 ): follower
                 for follower in followers
             }
-        for record in records:
-            thread_data = {}
-            if record._name != "discuss.channel":
-                # sudo: mail.thread - if mentionned in a non accessible thread, name is allowed
-                thread_data["name"] = record.sudo().display_name
-            if self.env[record._name]._original_module:
-                thread_data["module_icon"] = modules.module.get_module_icon(
-                    self.env[record._name]._original_module
-                )
-            if for_current_user and add_followers:
-                thread_data["selfFollower"] = Store.one(
-                    follower_by_record_and_partner.get((record, self.env.user.partner_id)),
-                    fields={"is_active": True, "partner": []},
+        record_fields = [
+            # sudo: mail.thread - if mentionned in a non accessible thread, name is allowed
+            Store.Attr(
+                "name",
+                lambda record: record.sudo().display_name,
+                predicate=lambda record: record._name != "discuss.channel",
```

Comment:
:facepalm:

Reply:
Not pretty, but some exceptions were needed to more forward without requiring to resolve them in the same PR. We don't want to keep them and we will look to clean them as soon as possible.

Reply:
Also this part of the code should eventually be in discuss folder as it's purely channel condition, it shouldn't be in standard message.

I'm not even sure if we still need/use the name/display name of thread from message.
It looks like something we only need maybe in inbox and maybe in notifications, I'll double check it.

Reply:
https://github.com/odoo/odoo/pull/190243